qutip/solver/options.py

```diff
@@ -0,0 +1,258 @@
+__all__ = ['SolverOptions',
+           'SolverResultsOptions', 'SolverOdeOptions',
+           'McOptions']
+
+from ..optionsclass import optionsclass
+import multiprocessing
+
+@optionsclass("solver")
+class SolverOptions:
+    """
+    Class of options for evolution solvers such as :func:`qutip.mesolve` and
+    :func:`qutip.mcsolve`. Options can be specified either as arguments to the
+    constructor::
+
+        opts = SolverOptions(progress_bar='enhanced', ...)
+
+    or by changing the class attributes after creation::
+
+        opts = SolverOptions()
+        opts['progress_bar'] = 'enhanced'
+
+    Returns options class to be used as options in evolution solvers.
+
+    The default can be changed by::
+
+        qutip.settings.solver['progress_bar'] = 'enhanced'
+
+    Options
+    -------
+    progress_bar : str {'text', 'enhanced', 'tqdm', ''}
+        How to present the solver progress.
+        'tqdm' use the python module of the same name and raise an error if
+        not installed.
+        True will result in 'text'.
+        Empty string or False will disable the bar.
```

Comment:
Nitpick: Could we put the explanations in the same order as the strings in the type?

What does `enhanced` do?

Reply:
Change it look:
text:
```
10.0%. Run time:   0.00s. Est. time left: 00:00:00:00
20.0%. Run time:   0.00s. Est. time left: 00:00:00:00
```
enhanced:
`[*******  30%               ] Elapsed 13.43s / Remaining 00:00:00:31`
Not sure how to describe them...